extensions/storage/README.md

```diff
@@ -0,0 +1,78 @@
+# Object Storage Extension Specification
+
+- **Title: Object Storage**
+- **Identifier: storage**
+- **Field Name Prefix: -**
+- **Scope: Item, Assets**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+- **Owner**: @davidraleigh
+
+This document explains the fields of the Object Storage Extension to a STAC Item and an Asset. This does not cover NFS solutions provided by PaaS cloud companies. Allows user to specify details related to cloud storage access and costs for assets and their STAC item.
+
+- [Example (Landsat 8)](examples/example-landsat8.json)
+- [JSON Schema](json-schema/schema.json)
+
+## Asset fields
+
+| Field Name  | Type   | Description |
+| ----------- | ------ | ----------- |
+| storage:cloud_platform        | string    | The [cloud provider](#providers) where data is stored |
+| storage:manager               | string    | The entity in charge of managing the data. |
+| storage:regions               | [string]  | The region(s) where the data is stored. Relevant to speed of access and inter region egress costs (as defined by PaaS provider) |
+| storage:bucket                | string    | The bucket for the asset, used along with object path |
```

Comment:
Azure doesn't have a single concept of a 'bucket', but has a 'storage account' and 'container'.

It seems like the `storage:bucket` and `storage:object_path` are redundant information for the asset `href`. Is the idea the `href` would be an http(s) url? I think if we were to drop the bucket and object_path properties, clients would be able to parse information from either the http endpoint or something like `s3://...`.

Reply:
I'm not a big fan of the redundant information by having `bucket` and `path` since it is already in href which encapsulates not only bucket, path, but also protocol.

Reply:
We use STAC a lot internally, so object_path and bucket are useful to those internal users who have access permissions to use them, but for customers there is an href that isn't constructed from bucket + object_path.

https://github.com/davidraleigh/stac-spec/blob/issue-896/extensions/storage/examples/example-nsl.json#L66
https://github.com/davidraleigh/stac-spec/blob/issue-896/extensions/storage/examples/example-nsl.json#L69